scipy/stats/tests/test_distributions.py

```diff
@@ -6143,6 +6143,26 @@ def test_fit_min(self):
         assert_allclose(res, ref)
 
 
+class TestDweibull:
+
+    @pytest.mark.parametrize("c, ref",
+                            [(0.3, 1.5502834334489721),
+                             (13, -0.33898771699248414)])
+    def test_entropy(self, c, ref):
+        # The reference values were calculated with mpmath:
+        # from mpmath import mp
+        # mp.dps = 150
+        #
+        # def entropy(c):
+        #     def pdf(x):
+        #         y = C * mp.fabs(x) ** (c - 1) * mp.exp(-mp.fabs(x) ** c) / 2
+        #         return y
```

Comment:
I recomputed with the `c = mp.mpf(c)`, but the values were the same. I'm pretty sure mpmath automatically casts to the `mpf` datatype when you make a calculation.

Reply:
It does, but it can't touch the calculations that are performed _before_ any operation is done by `mpmath`. For example, suppose you wanted:

```python3
mp.log(1/3)
```

This would lose precision compared to
```python3
mp.log(mp.one/3)
```

@dschmitz89's suggestion is a good one to avoid mistakes in the future.

Reply:
Thanks for clarifying. Sorry if I came off a bit dismissive, didn't mean to sound like I wasn't open to feedback, just trying to explain how I thought mpmath worked.